zilliqa/src/consensus.rs

```diff
@@ -364,6 +364,7 @@ impl Consensus {
             }
             // treat genesis as finalized
             consensus.set_finalized_view(latest_block_view)?;
+            return Ok(consensus);
```

Comment:
Return here if we are starting from genesis - the code below does not apply in this case